### ISSUE TITLE
[feature request] add option for `gum spin` to dump output on error

### ISSUE THREAD

**Issue by jackmac92** [charmbracelet/gum#55]
I think it would be great if `gum spin` could act as a prettier version of [chronic](https://manpages.debian.org/jessie/moreutils/chronic.1.en.html), which hides the output *unless* an error occurs

**Comment by maaslalani** [comment]
Hey @jackmac92, I think this is a really great idea. Let me/us think a little bit more about it / how to implement it (default behaviour vs opt-in, etc...)

**Comment by maaslalani** [comment]
Also, if you want a quick and dirty version of this functionality while we think about it, you can do this (on `gum` HEAD)

```
gum spin --show-output -- <command> >/dev/null
```

This will only display the `stderr` and not the stdout which should generally only output errors

**Comment by chrisfromredfin** [comment]
I'm piggybacking a little, but I'm realizing that this is close to what I want. I want to display a gum spinner while I run a long-running process _whose output I want/need_. That is, I'm trying something like:

`gum spin -s points --title "Fetching list of sites..." -- someLongCommand > output.txt`

What I really want is to display a spinner while someLongCommand writes _its_ output to a file. So, would there perhaps be a flag to also allow the output to pass through? Or to add a --output-to-file option or something?

**Comment by maaslalani** [comment]
@chrisfromredfin That should work but you need to use the `--show-output` to redirect the output to the file (also make sure you're on the latest version of gum)

**Comment by chrisfromredfin** [comment]
Yes but it also redirects the output of the spinner to the top of the file…

--
Christopher J. Wells
Redfin Solutions, LLC
office: 908-4-DRUPAL
fax: 207-292-1414
***@***.***
www.redfinsolutions.com

**Comment by maaslalani** [comment]
Ah sorry, I think the change that fixes this is not on `v0.4.0`.

@chrisfromredfin I think if you build on the latest code (including this commit: https://github.com/charmbracelet/gum/commit/4e61c120b0cbe8c2552fb961b7c5b3727497a798), the spinner will not be included in the file.

**Comment by chrisfromredfin** [comment]
Yes, just confirming that with a HEAD build it works! Excited for a new release. :)
